lib/client/weblogin.go

```diff
@@ -157,6 +164,62 @@ type SSHLogin struct {
 	// BindAddr is an optional host:port address to bind
 	// to for SSO login flows
 	BindAddr string
+	// Browser can be used to pass the name of a browser to override the system
+	// default (not currently implemented), or set to 'none' to suppress
+	// browser opening entirely.
+	Browser string
+	// RouteToCluster is an optional cluster name to route the response
+	// credentials to.
+	RouteToCluster string
+}
+
+// SSHLoginDirect contains SSH login parameters for direct (user/pass/OTP)
+// login.
+type SSHLoginDirect struct {
+	// ProxyAddr is the target proxy address
+	ProxyAddr string
+	// User is the login username.
+	User string
+	// User is the login password.
+	Password string
+	// User is the optional OTP token for the login.
+	OTPToken string
+	// PubKey is SSH public key to sign
+	PubKey []byte
+	// TTL is requested TTL of the client certificates
+	TTL time.Duration
+	// Insecure turns off verification for x509 target proxy
+	Insecure bool
+	// Pool is x509 cert pool to use for server certifcate verification
+	Pool *x509.CertPool
+	// Compatibility sets compatibility mode for SSH certificates
+	Compatibility string
+	// RouteToCluster is an optional cluster name to route the response
+	// credentials to.
+	RouteToCluster string
+}
+
+// SSHLoginU2F contains SSH login parameters for U2F login.
+type SSHLoginU2F struct {
+	// ProxyAddr is the target proxy address
+	ProxyAddr string
+	// User is the login username.
+	User string
+	// User is the login password.
+	Password string
+	// PubKey is SSH public key to sign
+	PubKey []byte
+	// TTL is requested TTL of the client certificates
+	TTL time.Duration
+	// Insecure turns off verification for x509 target proxy
+	Insecure bool
+	// Pool is x509 cert pool to use for server certifcate verification
+	Pool *x509.CertPool
+	// Compatibility sets compatibility mode for SSH certificates
+	Compatibility string
+	// RouteToCluster is an optional cluster name to route the response
+	// credentials to.
+	RouteToCluster string
```

Comment:
As this set of values is common to all `SSHLogin*` structs, is there any way for us to define it once and reuse it? Feels like a lot of boilerplate being copy/pasted with the same comments.

Reply:
Good idea, done